bundles/org.openhab.binding.solarforecast/src/main/java/org/openhab/binding/solarforecast/internal/actions/SolarForecastActions.java

```diff
@@ -0,0 +1,231 @@
+    private Optional<ThingHandler> thingHandler = Optional.empty();
+
+    @RuleAction(label = "@text/actionDayLabel", description = "@text/actionDayDesc")
+    public QuantityType<Energy> getDay(
+            @ActionInput(name = "localDate", label = "@text/actionInputDayLabel", description = "@text/actionInputDayDesc") LocalDate localDate,
+            String... args) {
+        if (thingHandler.isPresent()) {
+            List<SolarForecast> l = ((SolarForecastProvider) thingHandler.get()).getSolarForecasts();
+            if (!l.isEmpty()) {
+                QuantityType<Energy> measure = QuantityType.valueOf(0, Units.KILOWATT_HOUR);
+                for (Iterator<SolarForecast> iterator = l.iterator(); iterator.hasNext();) {
+                    SolarForecast solarForecast = iterator.next();
+                    State s = solarForecast.getDay(localDate, args);
+                    if (s instanceof QuantityType<?> quantityState) {
+                        measure = measure.add((QuantityType<Energy>) quantityState);
+                    } else {
+                        // break in case of failure getting values to avoid ambiguous values
+                        logger.trace("Ambiguous measure {} found for {} - return UNDEF", s, localDate);
+                        return Utils.getEnergyState(-1);
+                    }
+                }
+                return measure;
+            } else {
+                logger.trace("No forecasts found for {} - return UNDEF", localDate);
+                return Utils.getEnergyState(-1);
+            }
+        } else {
+            logger.trace("Handler missing - return UNDEF");
+            return Utils.getEnergyState(-1);
+        }
+    }
+
+    @RuleAction(label = "@text/actionPowerLabel", description = "@text/actionPowerDesc")
+    public QuantityType<Power> getPower(
+            @ActionInput(name = "timestamp", label = "@text/actionInputDateTimeLabel", description = "@text/actionInputDateTimeDesc") Instant timestamp,
+            String... args) {
+        if (thingHandler.isPresent()) {
+            List<SolarForecast> l = ((SolarForecastProvider) thingHandler.get()).getSolarForecasts();
+            if (!l.isEmpty()) {
+                QuantityType<Power> measure = QuantityType.valueOf(0, MetricPrefix.KILO(Units.WATT));
+                for (Iterator<SolarForecast> iterator = l.iterator(); iterator.hasNext();) {
+                    SolarForecast solarForecast = iterator.next();
+                    State s = solarForecast.getPower(timestamp, args);
+                    if (s instanceof QuantityType<?> quantityState) {
+                        measure = measure.add((QuantityType<Power>) quantityState);
+                    } else {
+                        // break in case of failure getting values to avoid ambiguous values
+                        logger.trace("Ambiguous measure {} found for {} - return UNDEF", s, timestamp);
+                        return Utils.getPowerState(-1);
```

Comment:
It seems this will actually return a `QuantityType` of -1 W, although it's documented that the method will return `UnDefType.UNDEF` is case of an error.

Please check others as well.

Reply:
All corrected, also in readme